api/registrations/serializers.py

```diff
@@ -545,13 +623,55 @@ def _find_orphan_files(self, registering, draft):
         from website.archiver.utils import find_selected_files
         files = find_selected_files(draft.registration_schema, draft.registration_metadata)
         orphan_files = []
-        for _, value in files.items():
+        for key, value in files.items():
             if 'extra' in value:
                 for file_metadata in value['extra']:
-                    if file_metadata['nodeId'] not in registering:
+                    if not self._is_attached_file_valid(file_metadata, registering):
                         orphan_files.append(file_metadata)
+
         return orphan_files
 
+    def _is_attached_file_valid(self, file_metadata, registering):
+        """
+        Validation of file information on registration_metadata.  Theoretically, the file information
+        on registration_responses does not have to be valid, so we enforce their accuracy here,
+        to ensure file links load properly.
+
+        Verifying that nodeId in the file_metadata is one of the files we're registering. Verify
+        that selectedFileName is the name of a file on the node.  Verify that the sha256 matches
+        a version on that file.
+
+        :param file_metadata - under "registration_metadata"
+        :param registering - node ids you are registering
+        :return boolean
+        """
+
+        node_id = file_metadata.get('nodeId')
+        if node_id not in registering:
+            return False
+
+        node = AbstractNode.load(node_id)
+        if not node:
+            # node in registration_metadata doesn't exist
+            return False
+
+        specified_sha = file_metadata.get('sha256', '')
+
+        file = node.files.filter(name=file_metadata.get('selectedFileName')).first()
+        if not file:
+            # file with this name does not exist on the node
+            return False
+
+        match = False
+        for version in file.versions.all():
+            if specified_sha == version.metadata.get('sha256'):
```

Comment:
this is reasonable -- but maybe implies the archiver should update the hash to match the version actually archived